x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlAuthenticator.java

```diff
@@ -236,6 +238,34 @@ private Assertion decrypt(EncryptedAssertion encrypted) {
         return attributes;
     }
 
+    private void checkAuthnStatement(List<AuthnStatement> authnStatements) {
+        if (authnStatements.size() != 1) {
+            throw samlException("SAML Assertion subject contains {} Authn Statements while exactly one was expected.", authnStatements
+                .size());
+        }
+        for (AuthnStatement authnStatement : authnStatements) {
```

Comment:
Since the list _must_ have exactly 1 element, it would be neater to have
```
final AuthnStatement authnStatement = authnStatement.get(0);
```
instead of a single-iteration loop